marconi-sidechain/doc/adr/01-sdd-computation.adoc

```diff
@@ -0,0 +1,203 @@
+Thus, it is a smaller data structure when compared to `LedgerState`.
+See https://github.com/input-output-hk/cardano-ledger/blob/master/docs/LedgerEvents.md for more information.
+====
+
+A container diagram of this solution would look like:
+
+image::http://www.plantuml.com/plantuml/png/jL7DRfmm4Bxp52ud8fL5gdhgAGsRIvKIb5NFaR67nOf_B7ieHLNVlRPtbmLfLVMI0vAPxsU-cRdtI1sEIcOVX6Pov406HEk_L9MZq-ueS1YxqODZHYDet36ZAYkfZfhI8vUYg-hFPHjBFxu_LOfw11TAhtLGKA71xUox-E_1IZEhA4eDiiPmABmr7qb-Q9y9N_3yZlpEiYLFuNIWl82aCucQ_6w5TJQ23zAmDz8wmqSczF52Un11bUGdMWYhHYqOHV1a2hFaeTXyQeXr1WqpCcYktlkkqC7d2VWHtEDFqBZl0d_rxmF_qk4MTN82OUDPA1fN9m9uVfLUS613GCzc2oEDtGEo8QHDZDBFcX58hK_HwoNwLdqxZmFAjt6Mvl48bpVvQcsqZrzFUpf1VtPVqQ-3C2fbYlFjyDoKBsszR7v5T__O_sQ8Frbsc3s2UdtyXGKaNKqTfzeG3JWPzxRu9zJzxsNlGVDHoRy0[]
+
+The benefits:
+
+* removal of the additional 10-16GB of memory needed to compute the `LedgerState` in `marconi-sidechain`
+* enables any chain-indexers (`cardano-db-sync`, `Kupo`, etc) to read the stored `LedgerEvent` in order to index various information like SDD and rewards without using a large amount of memory
+
+The drawbacks:
+
+* requires a change to `cardano-node` in order to store those LedgerEvents. This implies lots of discussions with the `cardano-node` team to accept such a change.
+* needs the SPO to resync the relay node from scratch in order to save the `LedgerEvent` on disk. However, such a resync will only be required once, even after an upgrade on `marconi-sidechain`.
+* need to maintain a separate fork of the `cardano-node` in the event of the node team not agreeing to merge those changes
+
+=== 3. Index SDD from the state query client of the node-to-client protocol
+
+This alternative solution takes advantage of the only way currently to get the SDD from a `cardano-node` by using the state query client of the node-to-client protocol.
+However, the query will only return the latest SDD given the slot from which the Cardano node is synced to.
+In order to take advantage of this functionality, we would need to change the deployment method of `marconi-sidechain` to something as follows:
+
+. The SPO needs to stop the relay node and delete the `cardano-node` database
+. Start `marconi-sidechain` which will wait for the node to start
+. Start the relay node
+. Once Marconi has noticed that the node has started syncing, it will start querying the local query state for the SDD for each new epoch and index it in the database.
+
+The container diagram is the same as the link:#compute-sdd-by-tracking-the-ledgerstate[Compute SDD by tracking the LedgerState] solution.
+
+The benefits:
+
+* no change to `cardano-node` are required
+* uses the existing infrastructure to get the required information
+
+The drawbacks:
+
+* needs the SPO to resync the relay node from scratch
+* any indexing logic or database schema change to Marconi would require reindexing the local node (thus deleting the node database)
```

Comment:
Just to clarify, is it because we won't be able to resume the SDD indexer?

Reply:
Let's say, you start the relay node and marconi-sidechain from scratch. Imagine you get a change in requirements from the Sidechain tribe or you optimize the database schema in marconi-sidechain for faster syncing time and query response time. 

In those scenarios, you would need to reindex every in `marconi-sidechain` from scratch, right? Therefore, with this solution, reindexing everything from scratch would require the SPO to delete the node database, and rerun everything.